addons/minio/template/base/install.sh

```diff
@@ -247,7 +271,344 @@ function allow_pvc_resize() {
+    fi
+
+    local minio_secret_key
+    minio_secret_key=$(kubernetes_secret_value "$MINIO_NAMESPACE" minio-credentials MINIO_SECRET_KEY)
+    if [ -z "$minio_secret_key" ]; then
+        bail "Failed to read minio access key"
+    fi
+
+    local minio_pod_ip
+    minio_pod_ip=$(kubectl get pods -n "$MINIO_NAMESPACE" -l app=minio --no-headers -o custom-columns=:.status.podIP | grep -v none | head -1)
+    if [ -z "$minio_pod_ip" ]; then
+        bail "Failed to determine minio pod ip address"
+    fi
+
+    minio_disable_minio_svc
+    if ! spinner_until 120 minio_svc_has_no_endpoints "$MINIO_NAMESPACE" "minio" ; then
+        minio_enable_minio_svc
+        bail "Timeout waiting for minio service to be decomissioned"
+    fi
+
+    if ! minio_create_fs_migration_deployment ; then
+        minio_enable_minio_svc
+        bail "Failed to start minio migration deployment"
+    fi
+
+    if ! migrate_object_store \
+        "$MINIO_NAMESPACE" \
+        "$minio_pod_ip:9000" \
+        "$minio_access_key" \
+        "$minio_secret_key" \
+        "minio-migrate-fs-backend" \
+        "$minio_access_key" \
+        "$minio_secret_key"
+    then
+        minio_enable_minio_svc
+        minio_destroy_fs_migration_deployment
+        bail "Failed to migrate data to minio migration deployment"
+    fi
+
+    # scale down minio and wait until it is out of service.
+    kubectl scale deployment -n "$MINIO_NAMESPACE" minio --replicas=0
+    if ! spinner_until 120 deployment_fully_updated minio "$MINIO_NAMESPACE"; then
+        minio_enable_minio_svc
+        minio_destroy_fs_migration_deployment
+        kubectl scale deployment -n "$MINIO_NAMESPACE" minio --replicas="$minio_replicas"
+        bail "Timeout scaling down minio deployment"
+    fi
+
+    # wait until are minio pods have been completely stopped.
+    if ! spinner_until 120 minio_pods_are_finished; then
+        minio_enable_minio_svc
+        minio_destroy_fs_migration_deployment
+        kubectl scale deployment -n "$MINIO_NAMESPACE" minio --replicas="$minio_replicas"
+        bail "Timeout waiting for minio pods to finish"
+    fi
+
+    if [ -z "$MINIO_HOSTPATH" ]; then
+        # get the pv in use by the minio deployment, we gonna need it later on when we swap the pvs.
+        local minio_pv
+        minio_pv=$(kubectl get pvc --no-headers -n "$MINIO_NAMESPACE" minio-pv-claim -o custom-columns=:.spec.volumeName)
+        if [ -z "$minio_pv" ]; then
+            minio_enable_minio_svc
+            minio_destroy_fs_migration_deployment
+            kubectl -n "$MINIO_NAMESPACE" scale deployment minio --replicas="$minio_replicas"
+            bail "Failed to find minio pv"
+        fi
+
+        # get the pv in use by the fs migration deployment, we gonna need it later on when we swap the pvs.
+        local migration_pv
+        migration_pv=$(kubectl get pvc --no-headers -n "$MINIO_NAMESPACE" minio-migrate-fs-backend-pv-claim -o custom-columns=:.spec.volumeName)
```

Comment:
we have a pattern of using -o jsonpath or --template. can you use that instead?

Reply:
fixed.